solr/core/src/java/org/apache/solr/security/PKIAuthenticationPlugin.java

```diff
@@ -153,7 +153,9 @@ public boolean doAuthenticate(
     PKIHeaderData headerData = null;
     String headerV2 = request.getHeader(HEADER_V2);
     String headerV1 = request.getHeader(HEADER);
-    if (headerV2 != null) {
+    if (headerV1 == null && headerV2 == null) {
```

Comment:
This makes the null check at L165 redundant.

Reply:
Yep - I considered removing it but ended up keeping it as I liked the explicitness/clarity.  But that's very "personal preference", and I'm happy to remove if you prefer?